projects/observability/src/shared/components/utils/d3/zoom/d3-zoom.ts

```diff
@@ -92,7 +88,7 @@ export abstract class D3Zoom<TContainer extends Element = Element, TTarget exten
     this.translateToRect(requestedRect);
   }
 
-  public panToRect(viewRect: ClientRect): void {
+  public panToRect(viewRect: Omit<DOMRect, 'x' | 'y' | 'toJSON'>): void {
```

Comment:
You replaced clientrect (was that a native type that's been expanded) but narrow it back in several places. Maybe add an alias for the dom rect without these fields?

Reply:
Yeah, ClientRect is deprecated now. DomRect is the preferred way. Updated with alias